### DISCUSSION TITLE
Ephemeral & Durable Consumer deleted on Server side not known in the C# Client library's Pull Subscription Object

### DISCUSSION THREAD

**Discussion by BSBalaji** [nats-io/nats.net.v1#902] [Q&A]
Ephemeral & Durable Consumer deleted on Server side not known in the C# Client library's Pull Subscription Object.

1. Create a Consumer (Ephemeral /Durable)
2. Start a thread that does (sub.PullNoWait() && sub.NextMessage(n)) on the subscription to get the new messages.
3. Receive some messages (optional)
4. Delete the consumer on the server / let it timeout and get removed
5. There is no exception in the next Pull (sub.PullNoWait() && sub.NextMessage(n))
6. Check the subscription state in the IJetStreamPullSubscription Object it says Valid.
Very similar tot his but without Bind Mode : https://github.com/nats-io/nats.net/discussions/627

**Comment by scottf** [comment]
Can you try using simplification? The simplification is aware of these things and provides much more robust consuming. See the `Simplification*` projects in https://github.com/nats-io/nats.net/tree/main/src/Samples
